delete.ps1

```diff
@@ -1,92 +1,285 @@
+        $SqlCmd.CommandText = $SqlQuery
+
+        # Set the data adapter
+        $SqlAdapter = [System.Data.SqlClient.SqlDataAdapter]::new()
+        $SqlAdapter.SelectCommand = $SqlCmd
+
+        # Set the output with returned data
+        $DataSet = [System.Data.DataSet]::new()
+        $null = $SqlAdapter.Fill($DataSet)
+
+        # Set the output with returned data
+        $Data.value = $DataSet.Tables[0] | Select-Object -Property * -ExcludeProperty RowError, RowState, Table, ItemArray, HasErrors
     }
-    $accountExists.Close()
-
-    # Check if user exists in the HelloID database
-    Switch ($lookupResult.count) {
-         0 {
-            #$queryAccount = $queryAccountCreate
-            Write-Verbose -Verbose -Message "Account record for '$($p.displayName)' with ExtUserId '$($account.ExtUserId)' does not exist in the HelloID database."
-            #Mark success here!
-        } 1 {
-            #$queryAccount = $queryAccountUpdate
-            Write-Verbose -Verbose -Message "Account record exists in the HelloID database. Soft deleting account record for person '$($p.displayName)' and ExtUserId '$($account.ExtUserId)'"
-        } default {
-            # TODO:"Implement audit stuff here
-            throw "Multiple ($($lookupResult.count)) account records found in the HelloID database for ExtUserId '$accountReference' : " + ($lookupResult | convertTo-Json)
+    catch {
+        $Data.Value = $null
+        Write-Error $_
+    }
+    finally {
+        if ($SqlConnection.State -eq "Open") {
+            $SqlConnection.close()
         }
+        Write-Verbose "Successfully disconnected from SQL database"
     }
+}
 
-     #Do not execute when running preview
-     if (-Not($dryRun -eq $True)) {
-
-        # Run account query
-        $sqlCmd = New-Object System.Data.SqlClient.SqlCommand
-        $sqlCmd.Connection = $sqlConnection
-        $sqlCmd.CommandText = $queryAccountSoftDelete
-        $account.Keys | Foreach-Object { $null = $sqlCmd.Parameters.Add("@" + $_, "$($account.Item($_))") }
-        $null = $SqlCmd.ExecuteNonQuery()
-        $success = $true
-        $auditMessage = "Soft deleted account record for person '$($p.displayName)' and ExtUserId '$($account.ExtUserId)' succesfully"
-        Write-Verbose -Verbose  -Message "Soft deleted account record for person '$($p.displayName)' and ExtUserId '$($account.ExtUserId)' succesfully"
-     }
-} catch {
-     $auditMessage = " not deleted succesfully: General error"
-     if (![string]::IsNullOrEmpty($_.ErrorDetails.Message)) {
-         Write-Verbose -Verbose -Message "Something went wrong $($_.ScriptStackTrace). Error message: '$($_.ErrorDetails.Message)'"
-         $auditMessage = " not deleted succesfully: '$($_.ErrorDetails.Message)'"
-     } else {
-         Write-Verbose -Verbose -Message "Something went wrong $($_.ScriptStackTrace). Error message: '$($_)'"
-         $auditMessage = " not deleted succesfully: '$($_)'"
-     }
+function ConvertTo-FlatObject {
```

Comment:
Do we still need this function now the fields with dots in it are gone in the field mapping?